src/impl/realm_coordinator.cpp

```diff
@@ -81,7 +81,14 @@ void RealmCoordinator::create_sync_session()
     if (m_sync_session)
         return;
 
-    m_sync_session = SyncManager::shared().get_session(m_config.path, *m_config.sync_config);
+    util::Optional<std::array<char, 64>> encryption_key;
+    if (m_config.encryption_key.data()) {
+        std::array<char, 64> key;
+        std::copy_n(m_config.encryption_key.begin(), key.size(), key.begin());
+        encryption_key = std::move(key);
+    }
+
+    m_sync_session = SyncManager::shared().get_session(m_config.path, *m_config.sync_config, std::move(encryption_key));
```

Comment:
`Optional<std::array<char, 64>>` is just a blob of 65 bytes, so "moving" it is still memcpy(). There's no way to have the newly constructed std::array just point to the data from the old std::array because std::array doesn't point at heap-allocated data.